exercises/protein-translation/protein_translation_test.go

```diff
@@ -102,71 +89,115 @@ func TestCodon(t *testing.T) {
+				test.input, ErrInvalidBase{}, err)
+		}
+		if actual != test.expected {
+			t.Fatalf("FAIL: Protein translation test: %s\nExpected: %s\nActual: %s",
+				test.input, test.expected, actual)
+		}
+		t.Logf("PASS: Protein translation test: %s", test.input)
+	}
+
+	for _, test := range codonStopCases {
+		actual, err := FromCodon(test.input)
+		if !errors.As(err, &ErrStop{}) {
+			t.Fatalf("FAIL: Protein translation test: %s\nExpected type error: %T\nActual type error: %T",
+				test.input, ErrStop{}, err)
+		}
+		if actual != test.expected {
+			t.Fatalf("FAIL: Protein translation test: %s\nExpected: %s\nActual: %s",
+				test.input, test.expected, actual)
+		}
+		t.Logf("PASS: Protein translation test: %s", test.input)
+	}
 }
 
 type rnaCase struct {
-	input         string
-	expected      []string
-	errorExpected error
+	input    string
+	expected []string
 }
 
-var proteinTestCases = []rnaCase{
+var proteinSuccessCases = []rnaCase{
 	{
 		"AUGUUUUCUUAAAUG",
 		[]string{"Methionine", "Phenylalanine", "Serine"},
-		nil,
 	},
 	{
 		"AUGUUUUGG",
 		[]string{"Methionine", "Phenylalanine", "Tryptophan"},
-		nil,
 	},
 	{
 		"AUGUUUUAA",
 		[]string{"Methionine", "Phenylalanine"},
-		nil,
 	},
 	{
 		"UGGUGUUAUUAAUGGUUU",
 		[]string{"Tryptophan", "Cysteine", "Tyrosine"},
-		nil,
 	},
+}
+
+var proteinInvalidBaseCases = []rnaCase{
 	{
 		"UGGAGAAUUAAUGGUUU",
 		[]string{"Tryptophan"},
-		ErrInvalidBase,
 	},
 }
 
 func TestProtein(t *testing.T) {
-	for _, test := range proteinTestCases {
+	for _, test := range proteinSuccessCases {
 		actual, err := FromRNA(test.input)
-		if test.errorExpected != nil {
-			if test.errorExpected != err {
-				t.Fatalf("FAIL: RNA translation test: %s\nExpected error: %q\nActual error: %q",
-					test.input, test.errorExpected, err)
-			}
-		} else if err != nil {
-			t.Fatalf("FAIL: RNA translation test: %s\nExpected: %s\nGot error: %q",
+		if err != nil {
+			t.Fatalf("FAIL: Protein translation test: %s\nExpected: %s\nGot error: %q",
 				test.input, test.expected, err)
 		}
 		if !reflect.DeepEqual(actual, test.expected) {
 			t.Fatalf("FAIL: RNA Translation test: %s\nExpected: %q\nActual %q", test.input, test.expected, actual)
 		}
 		t.Logf("PASS: RNA translation test: %s", test.input)
 	}
+	for _, test := range proteinInvalidBaseCases {
+		actual, err := FromRNA(test.input)
+		if !errors.As(err, &ErrInvalidBase{}) {
+			t.Fatalf("FAIL: Protein translation test: %s\nExpected type error: %T\nActual type error: %T",
+				test.input, ErrInvalidBase{}, err)
+		}
+		if !reflect.DeepEqual(actual, test.expected) {
+			t.Fatalf("FAIL: RNA Translation test: %s\nExpected: %q\nActual %q", test.input, test.expected, actual)
+		}
+		t.Logf("PASS: RNA translation test: %s", test.input)
+	}
 }
 
 func BenchmarkCodon(b *testing.B) {
-	for _, test := range codonTestCases {
+	for _, test := range codonSuccessCases {
+		for i := 0; i < b.N; i++ {
+			FromCodon(test.input)
+		}
+	}
+	for _, test := range codonInvalidBaseCases {
```

Comment:
There's no real value in benchmarking error returns, is there?

Reply:
I didn't understand this. As I know, the idea of this benchmarking function is to run all the scenarios without any check of the returns, but checking the time spent to run everything.

Reply:
Well, when you benchmark code, you're usually interested in its performance with valid data. If the data is invalid, of course we don't expect the function to do anything, so there's nothing to benchmark. So in benchmarks we usually only supply valid inputs.

Reply:
Now I understood what was your point. So dividing the scenarios as I did is actually better for the benchmark, and we can only run for the SuccessCases.